### ISSUE TITLE
Unreproducible XML error

### ISSUE THREAD

**Issue by RalfJung** [psteckler/ProofGeneral#80]
Occasionally, PG shows a message saying that there was an error in the XML communication with Coq. I have not yet managed to find a way to reproduce this (so I assume there is a race condition involved). I have however captured the communication log for one such occasion; you can find it attached:

[log.txt](https://github.com/psteckler/ProofGeneral/files/757934/log.txt)

**Comment by psteckler** [comment]
I don't see anything unusual in the XML.

Did you get the modal dialog suggesting that you should re-synch? Was the error preceded by interrupting the computation with C-c C-c?

**Comment by RalfJung** [comment]
> Did you get the modal dialog suggesting that you should re-synch?

Yes.

> Was the error preceded by interrupting the computation with C-c C-c?

No. All I did was start emacs with a particular file, and hit C-c C-RET in a particular spot. (It was the first empty line of `heap_lang/lib/barrier/proof.v` in Iris, and I got the same issue there three times in a row, but not again ever since.)

**Comment by psteckler** [comment]
Are you running on Windows, by any chance?

**Comment by RalfJung** [comment]
Nope, Linux. Debian testing amd64 to be more precise, with Coq installed via opam.

**Comment by psteckler** [comment]
There's probably a subtle bug in the code that feeds XML strings to the XML parser. 

I'll take a look at it again.

**Comment by psteckler** [comment]
If you set `proof-general-debug-messages` to non-nil when running PG, the offending XML will show up in the `*Messages*` buffer.

**Comment by RalfJung** [comment]
Is this something that I could set permanently in my `~/.emacs`, or will this create endless amounts of output and should not usually be enabled? If the latter is the case, it may be useful to print just the offending XML in case of an error, but not print anything if everything goes all right.

**Comment by psteckler** [comment]
This should be a rare error, which I've only seen when a long print operation is interrupted. It could be that the output is large.

I'll change it so the error is printed to messages, and does not need to be specially enabled.
